modules/siddhi-query-api/src/main/java/org/wso2/siddhi/query/api/expression/Expression.java

```diff
@@ -234,5 +239,63 @@ public static TimeConstant year(int i) {
             return year((long) i);
         }
 
+        public static Long timeToLong(String value) {
+            Pattern timeValuePattern = Pattern.compile("\\d+");
+            Pattern durationPattern = Pattern.compile("\\D+");
+            Matcher timeMatcher = timeValuePattern.matcher(value);
+            Matcher durationMatcher = durationPattern.matcher(value);
+            int timeValue;
+            TimePeriod.Duration duration;
+            if (timeMatcher.find() && durationMatcher.find()) {
+                duration = normalizeDuration(durationMatcher.group(0).trim());
+                timeValue = Integer.parseInt(timeMatcher.group(0));
+                switch (duration) {
+                    case SECONDS:
+                        return Expression.Time.sec(timeValue).value();
+                    case MINUTES:
+                        return Expression.Time.minute(timeValue).value();
+                    case HOURS:
+                        return Expression.Time.hour(timeValue).value();
+                    case DAYS:
+                        return Expression.Time.day(timeValue).value();
+                    case YEARS:
+                        return Expression.Time.year(timeValue).value();
+                    default:
+                        return Expression.Time.month(timeValue).value();
+                }
+            } else {
+                throw new SiddhiAppValidationException("Provided retention parameter cannot be identified. retention " +
+                        "period: " + value + ".");
+            }
+        }
+
+        public static TimePeriod.Duration normalizeDuration(String value) {
+            switch (value) {
```

Comment:
Shall we convert this to lowercase before starting the switch? The user might enter in upper case as well right? eg: Minutes

Reply:
Fixed by 6759878